### PR TITLE
[WIP] Improved PHPDoc comments

### PR DESCRIPTION
**Pull request by erayd**
## What
Improve PHPDoc comments as per #438.

## Why
As per #438.

### CODE REVIEW

**Comment by erayd** [comment]
Progress checklist for PHPDoc comments on public API entities:

## Methods
-  [ ] `Validator::__construct` - create a validator instance with a custom factory
-  [x] `Validator::validate` - main entry point for validation
-  [x] `Validator::check` - BC alias of `validate` (pre-`5.0.0`), does not pass `$value` by reference
-  [x] `Validator::coerce` - BC alias of `validate` (pre-`5.0.0`), sets `CHECK_MODE_COERCE_TYPES`
-  [ ] `Validator::getErrors` - get array of validation errors
-  [ ] `Validator::numErrors` - get number of validation errors
-  [ ] `Validator::getErrorMask` - get the error categories 
-  [ ] `Validator::isValid` - get overall validation status
-  [ ] `Validator::reset` - reset validation status
-  [ ] `Factory::__construct` - create a `Factory` instance with custom `SchemaStorage`, `UriRetriever` or `$checkmode`
 - [ ] `Factory::setConfig` - overwrite `$checkMode`
 - [ ] `Factory::getConfig` - get `$checkMode` or a filtered subset
 - [ ] `Factory::addConfig` - add options to `$checkMode`
 - [ ] `Factory::removeConfig` - remove options from `$checkMode`
 - [ ] `Factory::getSchemaStorage` - get the current `SchemaStorage` instance
 - [ ] `Factory::getUriRetriever` - get the current `UriRetriever` instance
 - [ ] `SchemaStorage::__construct` - create a `SchemaStorage` instance with custom `UriRetriever` or `UriResolver`
 - [ ] `SchemaStorage::addSchema` - add a schema to the storage instance by identifyng URI
 - [ ] `SchemaStorage::getSchema` - get a schema from the storage instance by identifying URI
 - [ ] `UriResolver::resolve` - resolve a URI to absolute
 - [ ] `UriRetriever::retrieve` - retrieve & return a decoded JSON schema object

## Class Constants
 - [ ] `Validator::SCHEMA_MEDIA_TYPE` - official JSON schema media type
 - [ ] `Validator::ERROR_*` - error categories (bitmask)
 - [ ] `Constraint::CHECK_MODE_*` - validation options (bitmask)

## Class Properties
 - [ ] `Validator::$factory` - dependency injection, state & instance management

## Data Formats
 - [ ] The array data returned by `Validator::getErrors()`

## Exceptions
 - [ ] `InvalidArgumentException`
 - [ ] `InvalidConfigException`
 - [ ] `InvalidSchemaException`
 - [ ] `InvalidSchemaMediaTypeException`
 - [ ] `InvalidSourceUriException`
 - [ ] `JsonDecodingException`
 - [ ] `ResourceNotFoundException`
 - [ ] `RuntimeException`
 - [ ] `UnresolvableJsonPointerException`
 - [ ] `UriResolverException`
 - [ ] `ValidationException`

File: src/JsonSchema/Validator.php [hunk 1]
```diff
@@ -32,13 +32,21 @@ class Validator extends BaseConstraint
...
-     * Validates the given data against the schema and returns an object containing the results
-     * Both the php object and the schema are supposed to be a result of a json_decode call.
-     * The validation works as defined by the schema proposal in http://json-schema.org.
+     * Validates the given data against the schema and returns an object containing the results.
+     *
+     * Both the value object and the schema are supposed to be the result of a json_decode call,
+     * or an equivalent value. The validation logic used is a non-compliant superset of the spec
+     * available at http://json-schema.org/ (versions prior to draft-06).
      *
      * Note that the first argument is passwd by reference, so you must pass in a variable.
```
Comment (shmax) [inline src/JsonSchema/Validator.php]: May as well fix the typo while you're in here ("passwd")

File: src/JsonSchema/Validator.php [hunk 2]
```diff
@@ -68,15 +76,34 @@ public function validate(&$value, $schema = null, $checkMode = null)
...
+     * Alias to validate(), to maintain backwards-compatibility with the previous API.
+     *
+     * Equivalent to calling validate() without setting any extra $checkMode options
+     *
+     * @api
+     *
+     * @param mixed $value  The value that should be validated against the schema - *not* a reference
+     * @param mixed $schema The schema to validate against
+     *
+     * @return int Bitwise list of error categories encountered during validation
```
Comment (shmax) [inline src/JsonSchema/Validator.php]: Maybe convey a little more information about the return, eg, tell us what value means no errors...

File: src/JsonSchema/Validator.php (hunk 1 above)
Comment (erayd) [inline src/JsonSchema/Validator.php]: Done.

File: src/JsonSchema/Validator.php (hunk 2 above)
Comment (erayd) [inline src/JsonSchema/Validator.php]: Done.

**Comment by bighappyface** [comment]
@erayd I am ready to get in on this.

I have put #392, #438 and #440 in progress in the v6.0.0 project. Is it fair to say that this PR will satisfy those two open issues?

Also, how would you prefer I contribute to this work?

**Pull request closed without merging**